waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/exchange/ExchangeExternalSignaturesController.kt

```diff
@@ -192,163 +234,182 @@ fun Application.exchangeExternalSignatures() = walletRoute {
+                    example(
+                        "W3C SD-JWT Verifiable Credential",
+                        ExchangeExternalSignaturesExamples.submitOid4vpRequestW3CSDJWTVCExample(),
+                    )
+                    example(
+                        "IETF SD-JWT Verifiable Credential",
+                        ExchangeExternalSignaturesExamples.submitOid4vpRequestIETFSDJWTVCExample(),
+                    )
                 }
             }
-            response(OpenAPICommons.usePresentationRequestResponse())
+
+            response(ExchangeOpenApiCommons.usePresentationRequestResponse())
         }) {
-            val req = call.receive<SubmitOID4VPRequest>()
-            logger.debug { "Request: $req" }
+            val walletService = getWalletService()
 
-            val authReq = req.resolvedAuthReq
-            val authResponseURL = authReq.responseUri
-                ?: authReq.redirectUri ?: throw AuthorizationError(
-                    authReq,
-                    AuthorizationErrorCode.invalid_request,
-                    "No response_uri or redirect_uri found on authorization request"
-                )
-            logger.debug { "Authorization response URL: $authResponseURL" }
-
-            val presentationSubmission = req.presentationSubmission
-            val presentedCredentialIdList = req.presentedCredentialIdList
-
-            val tokenResponse = TokenResponse.success(
-                vpToken = VpTokenParameter.fromJsonElement(req.signedVP.toJsonElement()),
-                presentationSubmission = presentationSubmission,
-                idToken = null,
-                state = authReq.state,
-            )
-
-            val formParams =
-                if (authReq.responseMode == ResponseMode.direct_post_jwt) {
-                    val encKey =
-                        authReq.clientMetadata?.jwks?.get("keys")?.jsonArray?.first { jwk ->
-                            JWK.parse(jwk.toString()).keyUse?.equals(KeyUse.ENCRYPTION) ?: false
-                        }?.jsonObject ?: throw Exception("No ephemeral reader key found")
-                    val ephemeralWalletKey =
-                        runBlocking { KeyManager.createKey(KeyGenerationRequest(keyType = KeyType.secp256r1)) }
-                    tokenResponse.toDirecPostJWTParameters(
-                        encKey,
-                        alg = authReq.clientMetadata!!.authorizationEncryptedResponseAlg!!,
-                        enc = authReq.clientMetadata!!.authorizationEncryptedResponseEnc!!,
-                        mapOf(
-                            "epk" to runBlocking { ephemeralWalletKey.getPublicKey().exportJWKObject() },
-                            "apu" to JsonPrimitive(Base64URL.encode(authReq.nonce).toString()),
-                            "apv" to JsonPrimitive(
-                                Base64URL.encode(authReq.nonce!!).toString()
-                            )
-                        )
+            runCatching {
```

Comment:
same here

however, this being the final step of the presentation, there probably has to be a distinction in the cause of the exception
and since there is an instance of incorrect logic mentioned, maybe have a separate ticket for solving it

Reply:
Refer to my above comment.

On the incorrect logic part, this does not only affect this handler, but also the regular `usePresentationRequest` handler. A separate work item has been already created to address the issue that responses obtained following the submission of Authorization Responses need to be reworked.